src/System Application/Test/AppSource Gallery/src/AppSourceProductManagerTest.Codeunit.al

```diff
@@ -0,0 +1,470 @@
+        ApplicationFamilyStore: Codeunit "Library - Variable Storage";
+        IsSaasStore: Codeunit "Library - Variable Storage";
+        KeyVaultStore: Codeunit "Library - Variable Storage";
+        TenantIdStore: Codeunit "Library - Variable Storage";
+        RestClientGetJsonStore: Codeunit "Library - Variable Storage";
+        CountryLetterCodeStore: Codeunit "Library - Variable Storage";
+
+    [Test]
+    procedure TestExtractAppIDFromUniqueProductIDReturnsExpectedAppId()
+    var
+        AppSourceProductManager: codeunit "AppSource Product Manager";
+        UniqueId: Text;
+        AppId: Text;
+    begin
+        // Given
+        UniqueId := 'PUBID.nav24spzoo1579516366010%7CAID.n24_test_transactability%7CPAPPID.0984da34-5ec1-4ac1-9575-b73fb2212327';
+
+        // When 
+        AppId := AppSourceProductManager.ExtractAppIDFromUniqueProductID(UniqueId);
+
+        // Then
+        Assert.AreEqual('0984da34-5ec1-4ac1-9575-b73fb2212327', AppId, 'Expected AppId to be extracted from UniqueId');
+    end;
+
+    [Test]
+    procedure TestExtractAppIDFromUniqueProductIDReturnsEmptyAppIdWhenNotValid()
+    var
+        AppSourceProductManager: codeunit "AppSource Product Manager";
+        UniqueId: Text;
+        AppId: Text;
+    begin
+        // Given
+        UniqueId := 'articentgroupllc1635512619530.ackee-ubuntu-18-04-minimal';
+
+        // When 
+        AppId := AppSourceProductManager.ExtractAppIDFromUniqueProductID(UniqueId);
+
+        // Then
+        Assert.AreEqual('', AppId, 'Expected AppId to be empty when not present in the UniqueId');
+    end;
+
+    [Test]
+    [HandlerFunctions('HyperlinkHandler')]
+    procedure TestOpenAppSourceHyperlink()
+    var
+        AppSourceProductManager: codeunit "AppSource Product Manager";
+        AppSourceProductManagerTest: Codeunit "AppSource Product Manager Test";
+    begin
+        Initialize();
+        // Initialize this copy
+        AppSourceProductManagerTest.Initialize();
+        // Override dependencies
+        AppSourceProductManager.SetDependencies(AppSourceProductManagerTest);
+
+        // Given
+        AppSourceProductManagerTest.AddToFormatRegionStore('da-DK');
+
+        // With handler expectation
+        HyperlinkStorage.Enqueue('https://appsource.microsoft.com/da-DK/marketplace/apps?product=dynamics-365-business-central');
+
+        // When 
+        AppSourceProductManager.OpenAppSource();
+
+        // Then
+        // Asserted in handler
+        AppSourceProductManagerTest.AssertCleanedUp();
+        AssertCleanedUp();
+    end;
+
+    [Test]
+    [HandlerFunctions('HyperlinkHandler')]
+    procedure TestOpenInAppSourceHyperlink()
+    var
+        AppSourceProductManager: codeunit "AppSource Product Manager";
+        AppSourceProductManagerTest: Codeunit "AppSource Product Manager Test";
+        UniqueId: Text;
+    begin
+        Initialize();
+        // Initialize this copy
+        AppSourceProductManagerTest.Initialize();
```

Comment:
I'm not sure I get why a copy of this test is used? Can't we just call the functions in this test directly? The only reason to make a copy of the test itself is for BindSubscription since we don't have a this operator.